### ISSUE TITLE
ICE: can't determine value for sysroot - FreeBSD

### ISSUE THREAD

**Issue by spamwax** [rust-lang/rust#41988]
rustc fails with the above message on FreeBSD 11.0 compiling any project/rust src.
As initially reported on [rustup repo](https://github.com/rust-lang-nursery/rustup.rs/issues/1110), this doesn't happen with the version of rust provided by OS package manager: rustc v1.16
However, installing the same version or later versions (tested up to 1.19) either through rustup or standalone installers produces the same ICE.

I tried this code:

```
$ rustup override set stable
info: using existing install for 'stable-x86_64-unknown-freebsd'
info: override toolchain for '/usr/home/hamid/src/test' set to 'stable-x86_64-unknown-freebsd'
  stable-x86_64-unknown-freebsd unchanged - rustc 1.17.0 (56124baa9 2017-04-24)
$ cargo build --verbose
error: failed to run `rustc` to learn about target-specific information
Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --target x86_64-unknown-freebsd` (exit code: 101)
--- stderr
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
thread 'rustc' panicked at '/checkout/src/librustc/session/filesearch.rs:164: can't determine value for sysroot', /checkout/src/librustc/session/mod.rs:787
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```
Same ICE happens when trying with different version of rustc either from `rustup` or standalone installers.

## Meta

```
$ rustc --version --verbose
rustc 1.17.0 (56124baa9 2017-04-24)
binary: rustc
commit-hash: 56124baa9e73f28c0709e59e74783cf234a978cf
commit-date: 2017-04-24
host: x86_64-unknown-freebsd
release: 1.17.0
LLVM version: 3.9
```

## Backtrace

```
$ RUST_BACKTRACE=1 cargo build --verbose
error: failed to run `rustc` to learn about target-specific information
Caused by:
  process didn't exit successfully: `rustc - --crate-name ___ --print=file-names --crate-type bin --crate-type rlib --target x86_64-unknown-freebsd` (exit code: 101)
--- stderr
error: internal compiler error: unexpected panic
note: the compiler unexpectedly panicked. this is a bug.
note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports
note: run with `RUST_BACKTRACE=1` for a backtrace
thread 'rustc' panicked at '/checkout/src/librustc/session/filesearch.rs:164: can't determine value for sysroot', /checkout/src/librustc/session/mod.rs:787
stack backtrace:
   0: <unknown>
   1: <unknown>
   2: <unknown>
   3: <unknown>
   4: <unknown>
   5: <unknown>
   6: <unknown>
   7: <unknown>
   8: <unknown>
   9: <unknown>
  10: <unknown>
  11: <unknown>
  12: <unknown>
  13: <unknown>
  14: <unknown>
  15: <unknown>
  16: <unknown>
  17: <unknown>
```

Output of `mount`:
```
$ mount
/dev/ada0p2 on / (ufs, local, journaled soft-updates)
devfs on /dev (devfs, local, multilabel)
tank on /tank (zfs, local, noatime, noexec, nfsv4acls)
tank/backup on /tank/backup (zfs, local, noatime, noexec, nfsv4acls)
home on /usr/home (zfs, local, noatime, nfsv4acls)
procfs on /proc (procfs, local)
```

**Comment by steveklabnik** [comment]
Does this still reproduce today?

**Comment by spamwax** [comment]
I don't have access to that system anymore, I'll try to test on a VM and see if it's reproducible

**Comment by Centril** [comment]
@spamwax How did your investigation go?

**Comment by spamwax** [comment]
The ICE seems to be fixed based on my test for a FreeBSD 12.1 running on VirtualBox

**Comment by spastorino** [comment]
Closing as fixed as per @spamwax comment.
